### ISSUE TITLE
Cannot type in new Workspace windows

### ISSUE THREAD

**Issue by j4yk** [MrModder/Autocompletion#43]
I updated Autocompletion during the last several days. Today I wanted to open a new Workspace window. When I hit the first key in it to type something, I get `MessageNotUnderstood: UndefinedObject>>model:` from Workspace>>createCompletionController. That method references the class ECWorkspaceController, which no longer exists in my image.

The preference is set so that it accesses the undefined binding, so nil becomes the receiver of the `model: self` message send.

**Comment by LeonMatthes** [comment]
Hm, that sounds like an issue where some of the old code wasn't deleted correctly during the update.
That has been an issue at times... Not sure that's something I can fix at my end.

The easiest fix would likely be to reinstall Autocompletion.

Otherwise you'll have to check which method on the Workspace wasn't correctly deleted during the update and delete it.

1. Go into the `Workspace` class and select the `*Autocompletion` category. If you're on the newest commit of Autocompletion master, there should only be two methods:
    1. `completionAdditionals`
    2. `guessTypeForName:`
2. Delete any additional methods that might still be in the `*Autocompletion` category. (Likely completionControllerClass or createCompletionController`

**Comment by LinqLover** [comment]
@j4yk Did you maybe install OCompletion somewhere in between? This could possibly have overridden any extension methods in a faulty way. Just guessing. :-)

**Comment by j4yk** [comment]
I didn't install OCompletion in between as far as I know, but it is possible that I had it installed before Autocompletion. The image is already a few years old. The Workspace>>createCompletionController method is indeed in the category *Ocompletion. I have removed it completely now.

**Comment by LeonMatthes** [comment]
@j4yk Did that solve the issue?
If so, please close the issue.

**Comment by j4yk** [comment]
Yes it works without the method, but I'm still not sure whether the upgrade path works as it should. If you install auto completion from the Squeak preference wizard into a fresh trunk image, Ocompletion gets installed. If I then load Autocompletion, the ECWorkspaceController remains in the image (category Ocompletion-ECView) and the Workspace>>#createCompletionController extension also remains in the *Ocompletion protocol. This still works (in contrast to my upgrade scenario where Autocompletion seemed to have adopted ECWorkspaceController before), but the system is not in the state you intend it to be, no?

**Comment by LeonMatthes** [comment]
Hm, you're right that's not the intended workflow.
The problem is really, that oCompletion and Autocompletion just aren't compatible anymore.
They both fork eCompletion and rely on extension methods with the same names.

As mentioned in #24 I've now removed the ability to revert to oCompletion entirely. So it's really best to remove oCompletion before installing Autocompletion.
It's very unfortunate that the default Squeak installer installs oCompletion though. I of course believe it should install Autocompletion instead :wink: 

In any case I've updated the README and readded an empty ECWorkspaceController so it should still work to install Autocompletion into an image that already has OCompletion installed.
